Simple Incremental/Assets/Scripts/HelperAttackEnemy.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using UnityEngine;
+
+[RequireComponent(typeof(EnemyTargeting))]
+public class HelperAttackEnemy : MonoBehaviour
+{
+    bool freeToAttack = true;
+    PlayerTarget target;
+    EnemyTargeting enemyTargetingRef;
+
+    private void Awake()
+    {
+        enemyTargetingRef = GetComponent<EnemyTargeting>();
+    }
+
+    private void Update()
+    {
+        if (freeToAttack)
+        {
+            StartCoroutine(StartAttackCycle());
+        }
+    }
+
+    private void OnEnable()
+    {
+        freeToAttack = true;
+    }
+
+    private IEnumerator StartAttackCycle()
+    {
+        target = enemyTargetingRef.FindATarget();
+
+        while (!freeToAttack || !target.gameObjectRef.activeSelf)
+        {
+            yield return null;
+        }
+        
+        while (target.gameObjectRef.activeSelf)

```

Comment:
The player has a Death Event. You can create a new "Player Died Event" explicitly for player deaths, and use that to flip your CoRoutine.

You are using gameObject activeSelf to check for liveliness. The gameObject may need to perform an animation, so it wouldn't be set as inactive on death immediately. You should have a public method that stops this while loop instead, and call it from a Game Event Listener that listens for Player Death events.